resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -86,7 +86,7 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   private val initialExecutors = SchedulerBackendUtils.getInitialTargetExecutorNumber(conf)
 
-  private val podAllocationInterval = conf.get(KUBERNETES_ALLOCATION_BATCH_DELAY)
+  private val podAllocationInterval = conf.getTimeAsMs(KUBERNETES_ALLOCATION_BATCH_DELAY.key)
```

Comment:
This should be just `conf.get(KUBERNETES_ALLOCATION_BATCH_DELAY)`. The config's unit is already ms.

Reply:
That was the previous state actually. Not sure what the best practice is here. @mridulm, thoughts? `getTimeAsMs` seemed better as it might protect us from changes in the config if at all that happens. It enforces the contract that it must be milliseconds, which is essential for the allocator to function correctly.

Reply:
`conf.get(KUBERNETES_ALLOCATION_BATCH_DELAY)` returns a `Long` if it's a time conf. That's how time configs are expected to be used.

You don't need `podAllocationInterval.toLong` later on like you had before.

Reply:
Done.

Reply:
I (incorrectly) assumed KUBERNETES_ALLOCATION_BATCH_DELAY was a String, and not ConfigEntry.
@vanzin's suggestion is much more elegant in comparison.